lib/ruby_lsp/requests/code_lens.rb

```diff
@@ -24,18 +24,18 @@ def provider
 
       sig do
         params(
-          global_state: GlobalState,
           uri: URI::Generic,
+          test_library: String,
           dispatcher: Prism::Dispatcher,
         ).void
       end
-      def initialize(global_state, uri, dispatcher)
+      def initialize(uri, test_library, dispatcher)
         @response_builder = T.let(
           ResponseBuilders::CollectionResponseBuilder[Interface::CodeLens].new,
           ResponseBuilders::CollectionResponseBuilder[Interface::CodeLens],
         )
         super()
-        Listeners::CodeLens.new(@response_builder, global_state, uri, dispatcher)
+        Listeners::CodeLens.new(@response_builder, test_library, uri, dispatcher)
 
         Addon.addons.each do |addon|
           addon.create_code_lens_listener(@response_builder, uri, dispatcher)
```

Comment:
@st0012 @vinistock I'm wondering how we should update this since currently ruby-lsp-rails checks `@global_state.test_library == "rails"`. What if we make `test_library` be an optional keyword argument? That would prevent breaking the API for other addons.

Reply:
This should be considered a breaking change anyway so I think changing args is expected?